app/helpers/katello/hosts_and_hostgroups_helper.rb

```diff
@@ -97,24 +103,28 @@ def relevant_organizations(host)
       end
     end
 
-    def lifecycle_environment_options(host, options = {})
+    # Generic method to provide a list of options in the UI
+    def content_options(host, selected_id, object_type, options = {})
       include_blank = options.fetch(:include_blank, nil)
       include_blank = '<option></option>' if include_blank == true #check for true specifically
-
-      selected_id = fetch_lifecycle_environment(host, options).try(:id)
       orgs = relevant_organizations(host)
       all_options = []
       orgs.each do |org|
-        env_options = ""
-        accessible_lifecycle_environments(org, host).each do |env|
-          selected = selected_id == env.id ? 'selected' : ''
-          env_options << %(<option value="#{env.id}" class="kt-env" #{selected}>#{h(env.name)}</option>)
+        content_object_options = ""
+        accessible_content_objects = if object_type == :lifecycle_environment
```

Comment:
Can we pass accessible_content_objects array itself into this method rather than object_type?

Reply:
Good point. Let's leave it as-is!